js/modules/modules.go

```diff
@@ -119,6 +119,17 @@ type VU interface {
 
 	// Runtime returns the goja.Runtime for the current VU
 	Runtime() *goja.Runtime
+
+	// Reserve needs a better name
+	// MUST only be called while absolutely certain that something will not let the iteration end between the start and
```

Comment:
That does sound like a too important constraint to only be in a comment (especially considering it's a public method)? Can we enforce that programmatically?

Reply:
I mean, it likely won't work if used in other places :shrug:. Detecting this will probably add too much additional overhead everywhere. 

In reality, I would expect `mi.vu.Reserve()` to only happen in a function call from js->(go extension/module) and before it return (back) to js in order to start doing something in a goroutine (see setTimeout). Given this in most cases it will be fine. 

To be honest, working with an event loop is always *not* exactly easy. You have many ways in which to cut yourself, and we already have ways that are really corrupting (and people do them) like people do use `goja.Runtime` from a goroutine they spawned even though it also currently executes some js code. 

The whole event loop is in practice a way to make this usage "safe", whether the programmer of an extension will  use it correctly is a different manner. I am not against trying to do some checks, but there isn't really an easy way to check "are we running code in js or are we in the event loop or in some other places". 

I can think of two ways to do something like that:
1. check that goroutine stacktrace through [runtime.Stack](https://pkg.go.dev/runtime#Stack) and check that somewhere in the stack the current call comes from the event loop for example. This probably will be heavy on the `Reserve` and will error out even when called in a goroutine while another goroutine is still blocking the loop. I don't know of a particular use for the latter, but it is still valid usage I guess, as long as something blocks the event loop.
2. have even more wrapping (a thing we are trying to do less of) around all go<->js interaction. This likely will need another breaking change for `modules.Module` interface and will likely have problems similar to the ones explained about common.Bind in #1802